source/nvda.pyw

```diff
@@ -77,6 +77,15 @@ import logHandler
 from logHandler import log
 import winUser
 import winKernel
+# Avoid a E402 'module level import not at top of file' warning,
+# because monkeypatches need to be applied first first
+from systemUtils import getProcessTokenOrigin  # noqa: E402
+
+try:
+	globalVars.appLogonSessionID = getProcessTokenOrigin(winKernel.GetCurrentProcess())

```

Comment:
An alternative implementation would be to create a\ function in `systemUtils` named `getAppLoginID` or similar which just retrieves logon ID of the current process first time it is called. If you're worried about performance you can cache the result similar to what is being done in `hasSyswow64Dir`. I see several advantages:
- The NVDA's logon ID is retrieved only when needed i.e. only in cases where user accesses Chromium 
- In the current approach if there is an error when getting NVDA's logon ID we won't know about it since logging is not yet ready and exception is swallowed.
- The NOQA comment would not be necessary :-D

Reply:
I like this idea.